pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -356,6 +424,51 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 	return nil
 }
 
+func (c *Controller) deleteWorkload(ctx context.Context, workloadName string) (err error) {
+	deploymentAvailableCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sDeployment%s", c.conditionsPrefix, operatorv1.OperatorStatusTypeAvailable))
+
+	workloadDegradedCondition := applyoperatorv1.OperatorCondition().
+		WithType(fmt.Sprintf("%sWorkloadDegraded", c.conditionsPrefix))
+
+	status := applyoperatorv1.OperatorStatus()
+	defer func() {
+		status = status.WithConditions(deploymentAvailableCondition, workloadDegradedCondition)
+		if applyError := c.operatorClient.ApplyOperatorStatus(ctx, c.controllerInstanceName, status); applyError != nil {
+			err = applyError
+		}
+	}()
+
+	if _, err := c.deploymentLister.Deployments(c.targetNamespace).Get(workloadName); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
Is the `Get()` call necessary before attempting the `Delete()` call? What does the extra call buy us?

Reply:
The `Get()` happens on the lister (cache), which means that if the deployment has already been deleted, it'll save us an extra API call to `Delete()`.

Reply:
Ah, I totally missed that the `Get()` would be on the cache - that makes sense :). Thanks!